Automattic/wp-calypso
# Stats: Use FixedNavigationHeader on summary pages.

#### Proposed Changes

Replaces the navigation header on the various Stats summary pages per #69179.

Before:
![StatsBefore](https://user-images.githubusercontent.com/40267301/197500483-91676d43-dca2-4ee1-84b5-fdd244de22f6.png)

After:
![StatsAfter](https://user-images.githubusercontent.com/40267301/197500504-350e5974-d004-4835-be44-ec648693458d.png)

#### Testing Instructions



* Spin up this branch locally.
* Visit `http://calypso.localhost:3000/stats/`
* Select a site from the Site Switcher.
* Click any of the module headers with a chevron. ie: Posts & pages or Countries.
* Confirm the summary page has a fixed navigation header. (try scrolling the content)
* Confirm layout is okay. (may need to adjust the spacing)
* Select a date from the chart view.
* Click the "Post & pages" heading and then click on "Traffic" in the navigation.
* Confirm the selected date is restored in the chart view.
* Test on the "Insights" tab as well.

### Known Issues
* Date is not restored if the range is changed in the summary view.
* ~~Date is not restored if a single post summary is viewed.~~ — Fixed
* ~~When viewing "Insights → Latest post summary", the back link is Traffic when is should be Insights. Not sure how to detect this so we can set it correctly.~~ — Fixed
* There's one instance of the old CakeHeader component in the code that I can't figure out how to visit it in the UI. I'm currently thinking it's not actually accessible.
* Still need to remove the existing instances of the CakeHeader component. Still in place as the new headers are behind a feature flag for now.

### Notes

Changing the header style like this introduces some behavioural changes that haven't been discussed. The old header made use of the browser history when clicked so if you clicked around in a summary (from 7 days to 30 days, for example), the back button would reverse those jumps. The new behaviour doesn't account for that (and probably shouldn't, considering it's a clearly labeled location).

Additionally, the old back button did not work correctly on all summary pages. Noting this behavioural change in case this was unintended when deciding on the new style fixed navigation.

#### Pre-merge Checklist






Related to #69179.
